core/cap-0046-06.md

```diff
@@ -198,20 +205,27 @@ compliance functionality.
 * Admin Interface
 *
 \******************************************************************************/
-// If "admin" is the administrator, burn "amount" from "from"
-fn burn(admin: Signature, nonce: BigInt, from: Identifier, amount: BigInt) -> Result<(), Error>;
+// If "admin" is the administrator and clawback is not disabled, clawback "amount" from "from".
+// The "amount" is burned. This function can be used on deauthorized balances. 
+fn clawback(admin: Signature, nonce: BigInt, from: Identifier, amount: u128) -> Result<(), Error>;
+
+// If "admin" is the administrator, deauthorize "id"
+fn deauthorize(admin: Signature, nonce: BigInt, id: Identifier) -> Result<(), Error>;
 
-// If "admin" is the administrator, freeze "id"
-fn freeze(admin: Signature, nonce: BigInt, id: Identifier) -> Result<(), Error>;
+// If "admin" is the administrator, authorize "id"
+fn authorize(admin: Signature, nonce: BigInt, id: Identifier) -> Result<(), Error>;
 
 // If "admin" is the administrator, mint "amount" to "to"
-fn mint(admin: Signature, nonce: BigInt, to: Identifier, amount: BigInt) -> Result<(), Error>;
+fn mint(admin: Signature, nonce: BigInt, to: Identifier, amount: u128) -> Result<(), Error>;
 
 // If "admin" is the administrator, set the administrator to "id"
 fn set_admin(admin: Signature, nonce: BigInt, new_admin: Identifier) -> Result<(), Error>;
 
-// If "admin" is the administrator, unfreeze "id"
-fn unfreeze(admin: Signature, nonce: BigInt, id: Identifier) -> Result<(), Error>;
+// If "admin" is the administrator, disable the clawback function for this contract 
+fn disable_clawback(admin: Signature, nonce: BigInt) -> Result<(), Error>;
```

Comment:
If we don't have `enable_clawback`, wouldn't it make more sense to have this in the initializer?

Reply:
I don't think I understand. Only the admin can disable clawback, while anyone can trigger initialization.

Reply:
The semantics between Soroban and classic are similar but not the same, which makes things a little confusing. On classic, the issuer can flip its own `AUTH_CLAWBACK_ENABLED_FLAG` account flag either way. If it's set, new trustlines will be clawback enabled. The issuer is then allowed to clear the trustline clawback flag, but cannot add the flag to a trustline.

I'm trying to do something similar here, but at the contract level since it's much simpler to deal with and I don't think anyone needs a per balance control for clawback.

Reply:
I updated the non-wrapper case to set clawback on `init`.